lib/datadog/tracing/contrib/graphql/patcher.rb

```diff
@@ -55,9 +57,34 @@ def patch_schema!(schema)
             end
           end
 
+          # Before https://github.com/rmosolgo/graphql-ruby/pull/4038 was introduced,
+          # we were left with incompatibilities between ddtrace 1.0 and older graphql gem versions.
+          def patch_legacy_gem!
+            return unless Gem::Version.new(::GraphQL::VERSION) <= Gem::Version.new('2.0.6')
+
+            ::GraphQL::Tracing::DataDogTracing.prepend(PatchLegacyGem)
+          end
+
           def get_option(option)
             Datadog.configuration.tracing[:graphql].get_option(option)
           end
+
+          # Patches the graphql gem to support ddtrace 1.0.
+          # This is not necessary in versions containing https://github.com/rmosolgo/graphql-ruby/pull/4038.
+          module PatchLegacyGem
+            # Ensure invocation to #trace method targets the new namespaced public API object,
+            # instead of the old global Datadog.trace.
+            # This is fixed in graphql > 2.0.6.
+            def tracer
+              options.fetch(:tracer, Datadog::Tracing) # GraphQL will invoke #trace on the returned object
+            end
```

Comment:
Actually this is fixed by https://github.com/rmosolgo/graphql-ruby/pull/3978 which was [in 2.0.3](https://github.com/rmosolgo/graphql-ruby/blob/master/CHANGELOG.md). So technically this would not be needed in `[2.0.3, 2.0.6]`.

(I do realize that the `fallback_transaction_name` fix is only in `2.0.7`, and so we need the patch anyway BUT I would suggest fixing or even deleting the comment, since it's a bit misleading, and a bad comment is worse than no comment :P)

Reply:
I had this comment mentioning version `2.0.3` at first, but purposefully updated it to `2.0.6` for a reason, but I don't recall why. I'll updated it to `2.0.3`.